src/wmtk/io/MeshReader.hpp

```diff
@@ -17,7 +17,7 @@ class MeshReader
     void read(Mesh& mesh);
 
 private:
-    const std::filesystem::path& m_filename;
+    std::filesystem::path m_filename;
```

Comment:
It does not need to be mutable. The important part is that it must not be a reference, otherwise the following code breaks at runtime:
```
MeshReader reader("test.hdf5");
reader.read(mesh1);
```
I just removed `const` because I once learned that members should never be `const`. I can add it back if you want.

Reply:
:shrug: i agree that const members can be a huge annoyance / why people would want do disallow them, but if the semantic is that a MeshReader is defined on a single file (and one MeshReader should only ever read that one file) then i'd rather this be const.